### ISSUE TITLE
Possible to do OR with match expressions?

### ISSUE THREAD

**Issue by voltagecontrolled** [irontec/sngrep#291]
Hello,

Is anyone aware of a way to do any logic with match expressions? Have tried pipe ("|") and comma (",") to separate expressions with no luck.

Example of what I am trying to do: 

`sngrep -i 1.1.1.1|2.2.2.2`

Was hoping that would match either 1.1.1.1 or 2.2.2.2. Tried with/without quotes, etc.

**Comment by davemidd** [comment]
Works for me:
`sngrep "1.1.1.1|2.2.2.2"`

**Comment by Kaian** [comment]
There are multiples ways to filter input in sngrep, but the earlier you do it, the best. 

TL;DR: Use BPF filters when possible. Try `sngrep host 1.1.1.1 or host 2.2.2.2` if you try to match packet addresses.

In order of performance:
 - BPF Fitler: Used in command line and passed to libpcap, this filter is the most powerful and avoids passing to sngrep not matching packets resulting in best performance. This filters can be also used in ngrep, tcpdump and any other libpcap based tool.

 ```
sngrep host 1.1.1.1 or host 2.2.2.2
```
   
- Matching expression: Used in command line and checked agains the first message of each SIP dialog. It accepts POSIX or PCRE regexps. This filters can also be used in ngrep and must be set before the BPF filter. It's useful if you're searching for a domain in the message or a special header in the initial message of a call.

```
sngrep "X-Header: My-Special-Value"  [optional-bpf-filter-here]
```
or 
```
sngrep "1.1.1.1\|2.2.2.2"
```
 - Display filter: Used in the first sngrep screen, it just filters the data displayed in the list. It accepts POSIX or PCRE regexps and matches data displayed in the screen.

 - Filter Screen: Like the previous one, but this applies to all messages contents of the data displayed in the list.

Regards,
Kaian

**Comment by voltagecontrolled** [comment]
> There are multiples ways to filter input in sngrep, but the earlier you do it, the best.

This information is incredibly helpful, thank you! And taking another read through the man page I should have caught the fact that I can use bpf filter logic!

Cheers,
Nate
